Kernel/FileSystem/TmpFS.cpp

```diff
@@ -0,0 +1,295 @@
+#include <Kernel/FileSystem/TmpFS.h>
+#include <Kernel/Process.h>
+#include <Kernel/Thread.h>
+
+NonnullRefPtr<TmpFS> TmpFS::create()
+{
+    return adopt(*new TmpFS);
+}
+
+TmpFS::TmpFS()
+{
+}
+
+TmpFS::~TmpFS()
+{
+}
+
+bool TmpFS::initialize()
+{
+    m_root_inode = TmpFSInode::create_root(*this);
+    return true;
+}
+
+InodeIdentifier TmpFS::root_inode() const
+{
+    ASSERT(!m_root_inode.is_null());
+    return m_root_inode->identifier();
+}
+
+void TmpFS::register_inode(NonnullRefPtr<TmpFSInode> inode)
```

Comment:
This can just be
```cpp
void TmpFS::register_inode(TmpFSInode& inode)
```
We're not passing ownership of the `TmpFSInode`, it just needs to be a live, non-null object.

Reply:
Ah, right, I was thinking of a different refcounting scheme (where refcount is stored by the smart pointer outside of the object)